frontend/src/metabase/visualizations/lib/LineAreaBarPostRender.js

```diff
@@ -1,7 +1,14 @@
+/* @flow weak */
+
 import d3 from "d3";
 
 import { clipPathReference } from "metabase/lib/dom";
 
+const X_LABEL_MIN_SPACING = 2; // minimum space we want to leave between labels
+const X_LABEL_ROTATE_90_THRESHOLD = 24; // tick width breakpoint for switching from 45° to 90°
+const X_LABEL_HIDE_THRESHOLD = 12; // tick width breakpoint for hiding labels entirely
+const X_LABEL_MAX_LABEL_HEIGHT_RATIO = 0.7; // percent rotated labels are allowed to take
```

Comment:
My hunch is that 70% of the render area being taken up by labels is too high, but I'm down to wait until we see how this works on our internal instances before tweaking it further.